### PR TITLE
Make the source Mypy compliant

### LINKED ISSUE #101: Implement stylechecks
Stylecheck (black, isort, flake8 , mypy) on current code as well as setup automate style check for future pull request

### PR DESCRIPTION
**Pull request by gwwatkin**
Started from @alexnguyenn 's mypy reformat branch.

Checked with:
```
$ mypy --ignore-missing-imports lsqecc
$ mypy --ignore-missing-imports webapi
```

### CODE REVIEW

File: lsqecc/pauli_rotations/circuit.py [hunk 1]
```diff
@@ -89,11 +89,13 @@ def apply_transformation(self, start_index: int = 0) -> None:
...
 
         for i in range(start_index, len(self)):
-            if isinstance(self.ops[i], PauliRotation) and self.ops[i].rotation_amount in {
-                Fraction(1, 4),
-                Fraction(-1, 4),
-            }:
-                quarter_rotation.append(i)
+            if isinstance(self.ops[i], PauliRotation):
+                temp = cast(PauliRotation, self.ops[i])
+                if temp.rotation_amount in {
```
Comment (alexnguyenn) [inline lsqecc/pauli_rotations/circuit.py]:
```suggestion
                if cast(PauliRotation, self.ops[i]).rotation_amount in {
```
Just fixing my code here 😅

File: lsqecc/pauli_rotations/circuit.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import copy
 import pyzx as zx
 from fractions import Fraction
-from typing import List, cast
+from typing import List, cast, Type
```
Comment (alexnguyenn) [inline lsqecc/pauli_rotations/circuit.py]:
```suggestion
from typing import List, cast
```
Seems like `Type` was not used here?

**Review by alexnguyenn — COMMENTED** [review]
Also in `lsqecc/simulation/conditional_operation_control.py:24`, should change that to `NotImplementedError` as well

File: lsqecc/pauli_rotations/circuit.py (hunk 2 above)
Comment (gwwatkin) [reply lsqecc/pauli_rotations/circuit.py]: Mmmm I wonder why black diddn't remove it

File: lsqecc/pauli_rotations/circuit.py (hunk 1 above)
Comment (gwwatkin) [reply lsqecc/pauli_rotations/circuit.py]: Yep :+1:  I remember us talking about `cast`

**Review by alexnguyenn — APPROVED** [review]

**Pull request merged**